app/scripts/services/validateRegistrant.js

```diff
@@ -4,52 +4,36 @@ angular.module('confRegistrationWebApp')
   .service('validateRegistrant', function validateRegistrant() {
 
     var blockVisibleRuleCheck = function (block, registrant) {
-      var returnValueAND = true;
-      var returnValueOR = false;
       var answers = registrant.answers;
-      var ruleOperand = block.content && block.content !== '' && !angular.isUndefined(block.content.ruleoperand) ? block.content.ruleoperand : 'AND';
+      var ruleOperand = block.content && block.content.ruleoperand ? block.content.ruleoperand : 'AND';
+      var ruleArray = [];
 
-      if (ruleOperand === 'AND') {
-        angular.forEach(block.rules, function (rule) {
-          var answer = _.find(answers, { blockId: rule.parentBlockId });
```

Comment:
Guess you didn't like my code... You could collapse those `if`s together in mine if that would make it cleaner. I wrote it pretty quickly. I guess this is mostly a stylistic discussion. They all work. I'll let you decide which implementation you like best.